File: src/codeforces/div2/round604/BeautifulNumbers.java
package codeforces.exercises;

import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.StringTokenizer;

public class BeautifulNumbers {

    public static void main(String[] args) throws IOException {
        FastReader br = new FastReader();
        int t = br.nextInt();

        for (int i = 0; i < t; i++) {
            int n = br.nextInt();
            int[] pos = new int[n + 1];

            for (int j = 0; j < n; j++) {
                pos[br.nextInt()] = j;
            }
            System.out.println(findBeautifulNumbers(pos));
        }
    }

    private static String findBeautifulNumbers(int[] pos){
        int currMax = 0;
        int currMin = Integer.MAX_VALUE;
        String res = "";

        for (int i = 1; i < pos.length; i++) {
            currMax = Math.max(currMax, pos[i]);
            currMin = Math.min(currMin, pos[i]);

            if (currMax - currMin == i - 1){
                res += '1';
            }else {
                res += '0';
            }
        }
        return res;
    }

    private static class FastReader {
        BufferedReader br;
        StringTokenizer st;

        public FastReader() {
            br = new BufferedReader(new InputStreamReader(System.in));
        }

        String next() {
            while (st == null || !st.hasMoreElements()) {
                try {
                    st = new StringTokenizer(br.readLine());
                } catch (IOException e) {
                    e.printStackTrace();
                }
            }
            return st.nextToken();
        }

        int nextInt() {
            return Integer.parseInt(next());
        }

        long nextLong() {
            return Long.parseLong(next());
        }

        double nextDouble() {
            return Double.parseDouble(next());
        }

        String nextLine() {
            String str = "";
            try {
                str = br.readLine();
            } catch (IOException e) {
                e.printStackTrace();
            }
            return str;
        }
    }
}

File: src/codeforces/exercises/BeautifulNumbers.kt
package codeforces.exercises

import java.io.BufferedReader
import java.io.IOException
import java.io.InputStreamReader
import java.util.*
import kotlin.math.max
import kotlin.math.min


/**
 * https://codeforces.com/problemset/problem/1265/B
 */
fun main() {
    val br = FastReader()
    val t = br.nextInt()

    for (i in 1..t){
        val n = br.nextInt()
        val pos = Array(n + 1){_ -> 0}

        for (i in 0 until n){
            pos[br.nextInt()] = i
        }

        println(findBeautifulNumbers(pos))
    }

}

fun findBeautifulNumbers(pos: Array<Int>): String{
    tailrec fun go(i: Int, minPos: Int, maxPos: Int, res: String): String =
        if (i == pos.size) res
        else{
            val newMax = max(maxPos, pos[i])
            val newMin = min(minPos, pos[i])
            if (newMax - newMin == i - 1){
                go(i + 1, newMin, newMax, res + '1')
            }else{
                go(i + 1, newMin, newMax, res + '0')
            }
        }
    return go(1, Int.MAX_VALUE, 0, "")
}

internal class FastReader {
    var br: BufferedReader
    var st: StringTokenizer? = null
    operator fun next(): String {
        while (st == null || !st!!.hasMoreElements()) {
            try {
                st = StringTokenizer(br.readLine())
            } catch (e: IOException) {
                e.printStackTrace()
            }
        }
        return st!!.nextToken()
    }

    fun nextInt(): Int {
        return next().toInt()
    }

    fun nextLong(): Long {
        return next().toLong()
    }

    fun nextDouble(): Double {
        return next().toDouble()
    }

    fun nextLine(): String {
        var str = ""
        try {
            str = br.readLine()
        } catch (e: IOException) {
            e.printStackTrace()
        }
        return str
    }

    init {
        br = BufferedReader(InputStreamReader(System.`in`))
    }
}